src/client/pkg/grpcutil/server.go

```diff
@@ -21,9 +21,10 @@ var (
 
 // ServeOptions represent optional fields for serving.
 type ServeOptions struct {
-	Version    *versionpb.Version
-	MaxMsgSize int
-	Cancel     chan struct{}
+	Version     *versionpb.Version
+	ClusterInfo *versionpb.ClusterInfo
+	MaxMsgSize  int
+	Cancel      chan struct{}
 }
```

Comment:
Can we just have the APIServer get the cluster ID out of etcd rather than passing it in. That seems a bit simpler and a bit more resilient in case we add a way to change the ID or something in the future.